### PR TITLE
Upgrade Ray version for Autopilot; shrink worker resource allocation

### PR DESCRIPTION
**Pull request by artemvmin**
- Ray version 2.7.1 appears to fix some Ray Autoscaler bugs
- Lower resource allocation to better utilize node capacity (recommended 1:1 matching between k8s and Ray nodes):
  - Head: `c3-standard-4`
  - CPU worker:`c3-standard-8`
  - GPU worker:`g2-standard-8`
- Pretty print Ray job info in Jupyter notebook
- Fix README wording

### CODE REVIEW

File: modules/kuberay-cluster/kuberay-autopilot-values.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 image:
   # Replace this with your own image if needed.
   repository: rayproject/ray
-  tag: 2.6.1-py310-gpu
+  tag: 2.9.3-py310-gpu
```
Comment (imreddy13) [inline modules/kuberay-cluster/kuberay-autopilot-values.yaml:25]: should we bump it for Standard as well to keep things consistent?

Comment (artemvmin) [reply modules/kuberay-cluster/kuberay-autopilot-values.yaml:25]: That's going to be a larger, possibly breaking change, that affects multiple applications. I would rather do that in a separate PR.

**Review by imreddy13 — APPROVED** [review]

File: modules/kuberay-cluster/kuberay-autopilot-values.yaml (hunk 1 above)
Comment (saikat-royc) [reply modules/kuberay-cluster/kuberay-autopilot-values.yaml:25]: this needs a corresponding change to the notebook where we pip install ray ==2.7.x . We need to align the ray versions ?

Has this change been validate with the notebook e2e?

Comment (imreddy13) [reply modules/kuberay-cluster/kuberay-autopilot-values.yaml:25]: +1 to what Saikat said, my bad for missing this in the review. the notebook uses the same ray version as the standard kuberay config, does this work for AP?

**Pull request merged**